olp-cpp-sdk-dataservice-write/test/DataserviceWriteTestMain.cpp

```diff
@@ -20,7 +20,14 @@
 #include <gtest/gtest.h>
 #include "testutils/CustomParameters.hpp"
 
+#include <olp/core/client/OlpClientSettingsFactory.h>
+
+std::shared_ptr<olp::http::Network> g_network;
+
 int main(int argc, char** argv) {
+  g_network = olp::client::OlpClientSettingsFactory::
```

Comment:
There are some tests, which test parallel execution (like `StreamLayerClientOnlineTest.ConcurrentPublishDifferentIngestApi`). 

By executing requests in parallel and creating several instances of network, we can check whether we do not have any data race inside network implementations, so, I do not recommend to create shared instance of network per thread.

Reply:
this network instance is created before any thread is spawned, it's shared across all threads.

Reply:
Looks like indeed it was the original intention behind this test, but it helped me to find my problem. :) I agree that we might need indeed have such tests for `Network` itself.